Lib/cu2qu/ufo.py

```diff
@@ -137,30 +137,67 @@ def _segments_to_quadratic(segments, max_err, stats):
     return [('qcurve', p) for p in new_points]
 
 
-def _fonts_to_quadratic(fonts, max_err, reverse_direction, stats):
-    """Do the actual conversion of fonts, after arguments have been set up."""
+def _glyphs_to_quadratic(glyphs, max_err, reverse_direction, stats):
+    """Do the actual conversion of a set of compatible glyphs, after arguments
+    have been set up.
+    Return True if the glyphs were modified, else return False.
+    """
 
-    for glyphs in zip(*fonts):
-        name = glyphs[0].name
-        assert all(g.name == name for g in glyphs), 'Incompatible fonts'
+    glyphs_modified = False
 
-        segments_by_location = zip(*[_get_segments(g) for g in glyphs])
-        if not any(segments_by_location):
-            continue
+    segments_by_location = zip(*[_get_segments(g) for g in glyphs])
+    if not any(segments_by_location):
+        return glyphs_modified
 
-        new_segments_by_location = []
-        for segments in segments_by_location:
-            tag = segments[0][0]
-            assert all(s[0] == tag for s in segments[1:]), (
-                'Incompatible glyphs "%s"' % name)
-            if tag == 'curve':
-                segments = _segments_to_quadratic(segments, max_err, stats)
-            new_segments_by_location.append(segments)
+    new_segments_by_location = []
+    for segments in segments_by_location:
+        tag = segments[0][0]
+        assert all(s[0] == tag for s in segments[1:]), (
+            'Incompatible glyphs: ' + ", ".join(
+                set(repr(g.name) for g in glyphs)))
+        if tag == 'curve':
+            segments = _segments_to_quadratic(segments, max_err, stats)
+            glyphs_modified = True
+        new_segments_by_location.append(segments)
 
+    if glyphs_modified:
         new_segments_by_glyph = zip(*new_segments_by_location)
         for glyph, new_segments in zip(glyphs, new_segments_by_glyph):
             _set_segments(glyph, new_segments, reverse_direction)
 
+    return glyphs_modified
+
+
+def glyphs_to_quadratic(
+        glyphs, max_err=None, reverse_direction=False, stats=None):
+    """Convert the curves of a set of compatible of glyphs to quadratic.
+
+    All curves will be converted to quadratic at once, ensuring interpolation
+    compatibility. If this is not required, calling glyphs_to_quadratic with one
+    glyph at a time may yield slightly more optimized results.
+
+    Return True if glyphs were modified, else return False.
+    """
+    if stats is None:
+        stats = {}
+
+    if not max_err:
+        # assume 1000 is the default UPEM
+        max_err = DEFAULT_MAX_ERR * 1000
+
+    if isinstance(max_err, (list, tuple)):
+        max_errors = max_err
+    else:
+        max_errors = [max_err] * len(glyphs)
+
+    num_glyphs = len(glyphs)
+    assert len(max_errors) == num_glyphs
+
+    modified = _glyphs_to_quadratic(
+        glyphs, max_errors, reverse_direction, stats)
+
+    return modified
```

Comment:
These last five lines can be reduced to two lines:
```
assert len(max_errors) == len(glyphs)
return _glyphs_to_quadratic(glyphs, max_errors, reverse_direction, stats)
```

and we can make a corresponding change in fonts_to_quadratic to remove one line.